src/main/java/decodes/db/DatabaseIO.java

```diff
@@ -37,47 +48,45 @@ public abstract class DatabaseIO
 	  @return the DatabaseIO object
 	  @throws DatabaseException if type unrecognized or location is invalid.
 	*/
-	public static final DatabaseIO makeDatabaseIO(int type, String location)
-		throws DatabaseException
+	public static final DatabaseIO makeDatabaseIO(int type, String location) throws DatabaseException
+	{
+		return null;
+	}
+
+	public static final DatabaseIO makeDatabaseIO(DecodesSettings settings) throws DatabaseException
+	{
+		return makeDatabaseIO(settings, settings.editDatabaseLocation);
+	}
+
+	public static final DatabaseIO makeDatabaseIO(DecodesSettings settings, String locOverride) throws DatabaseException
 	{
 		ResourceFactory.instance().initDbResources();
 		
-		try 
+		try
 		{
-			if (type == DecodesSettings.DB_XML)
-				return new XmlDatabaseIO(location);
+			final String location = locOverride.startsWith("jdbc:") ? locOverride : "jdbc:xml:" + locOverride;
+			final int type = settings.editDatabaseTypeCode;
+
+			AuthSource auth = AuthSourceService.getFromString(settings.DbAuthFile);
```

Comment:
We may neeed a AuthSource for the XML database?   Testing with initial install gives:

```
C:\test\OPENDCS-auth\bin>dbedit
Exception in thread "main" java.lang.NullPointerException
        at org.opendcs.authentication.AuthSourceService.getFromString(AuthSourceService.java:35)
        at org.opendcs.database.DatabaseService.getDatabaseFor(DatabaseService.java:24)
        at decodes.dbeditor.DecodesDbEditor.main(DecodesDbEditor.java:132)
```

Reply:
That would be the NoOp datasource that you can see in the tests. Though it should've tried to default to that.